gitlab/client.py

```diff
@@ -1130,6 +1134,29 @@ def _query(
                 next_url = requests.utils.parse_header_links(result.headers["links"])[
                     0
                 ]["url"]
+            # if the next url is different with user provided server URL
+            # then give a warning it may because of misconfiguration
+            # but if the option to fix provided then just reconstruct it
+            if not next_url.startswith(self._gl.url):
+                search_api_url = re.search(r"(^.*?/api)", next_url)
+                if search_api_url:
+                    next_api_url = search_api_url.group(1)
+                    if self._gl.persist_base_url:
+                        next_url = next_url.replace(
+                            next_api_url, f"{self._gl._base_url}/api"
+                        )
+                    else:
+                        utils.warn(
+                            message=(
+                                f"The base url of the returned next page got "
+                                f"different with the user provided "
+                                f"{self._gl.url}/api* ~> {next_api_url}*, "
+                                f"since this may can lead to unexpected behaviour. "
+                                f"set argument persist_base_url to True for "
+                                f"resonctruct it with the origin one."
```

Comment:
Maybe:

```suggestion
                                f"The base URL in the server response"
                                f"differs from the user-provided base URL "
                                f"({self._gl.url}/api/ -> {next_api_url}/). "
                                f"This may lead to unexpected behavior and  "
                                f"broken pagination. Use `keep_base_url=True` "
                                f"when initializing the Gitlab instance "
                                f"to follow the user-provided base URL."
```

Reply:
done, set as suggested